system/src/system_network_diagnostics.cpp

```diff
@@ -308,6 +308,27 @@ class NetworkCellularCellGlobalIdentityMobileNetworkCodeDiagnosticData
     }
 } g_networkCellularCellGlobalIdentityMobileNetworkCodeDiagnosticData;
 
+class NetworkCellularCellGlobalIdentityTwoDigitMobileNetworkCodeDiagnosticData
+    : public AbstractIntegerDiagnosticData
+{
+public:
+    NetworkCellularCellGlobalIdentityTwoDigitMobileNetworkCodeDiagnosticData()
+        : AbstractIntegerDiagnosticData(
+              DIAG_ID_NETWORK_CELLULAR_CELL_GLOBAL_IDENTITY_TWO_DIGIT_MOBILE_NETWORK_CODE,
+              DIAG_NAME_NETWORK_CELLULAR_CELL_GLOBAL_IDENTITY_TWO_DIGIT_MOBILE_NETWORK_CODE)
+    {
+    }
+
+    virtual int get(IntType& val)
+    {
+        CellularGlobalIdentity cgi;
+        cellular_global_identity(&cgi, nullptr);
+        val = static_cast<IntType>(cgi.two_digit_mobile_network_code);
```

Comment:
Instead of introducing a new diagnostic type to just indicate whether the code is 2 or 3 digits, we could potentially add some kind of a flag to the current reported MNC value. Perhaps offset 2-digit values by something like a `10000` or set the most significant bit? cc @m-mcgowan

Reply:
When we initially discussed this, I thought all these are numbers, so to simplify development I suggested using current `IntType`. Perhaps we should revisit that decision, and this could potentially mean that we do need a string diagnostic type, so perhaps we should introduce that instead. :man_shrugging:

Reply:
I believe we arrived at a successful compromise during scrum.